project-addons/custom_partner/partner.py

```diff
@@ -30,12 +31,33 @@
 from datetime import datetime, timedelta
 from openerp.report import report_sxw
 
+
 class ResPartnerInvoiceType(models.Model):
     _name = 'res.partner.invoice.type'
 
     name = fields.Char('Name', required=True)
 
 
+class ResPartnerPrint(osv.Model):
+    _inherit = "res.partner"
+
+    @api.one
+    def _get_average_margin(self):
+        self.env.cr.execute(""" SELECT sum(sol.margin) / count(sol.id)
+                            FROM account_invoice ai
+                            JOIN account_invoice_line ail ON ail.invoice_id = ai.id
+                            LEFT JOIN sale_order_line_invoice_rel solir on solir.invoice_id = ail.id
+                            LEFT JOIN sale_order_line sol on sol.id = solir.order_line_id
+                                WHERE sol.order_partner_id = %d 
+                                AND ai.state IN ('paid', 'history')
+                                AND ai.date_invoice BETWEEN 
+                                    (CURRENT_DATE - interval '3 Month')::timestamp::date AND CURRENT_DATE """ % self.id)
+        res = self.env.cr.fetchone()[0]
```

Comment:
No os acostumbréis a hacer queries, en este caso no hay mucho problema aunque os estaría sumando facturas de pruebas si las tuviera, pero en otros si puede existir ya que con las queries se salta todas las reglas de permisos.